lightning/src/ln/channel.rs

```diff
@@ -1162,40 +1164,72 @@ impl<'a, SP: Deref> ChannelPhase<SP> where
 
 /// A top-level channel struct, containing a channel phase
 pub(super) struct ChannelWrapper<SP: Deref> where SP::Target: SignerProvider {
-	phase: ChannelPhase<SP>,
+	/// The inner channel phase
+	/// Option is used to facilitate in-place replacement (see e.g. move_v2_to_funded),
+	/// but it is never None, ensured in new() and set_phase()
+	phase: Option<ChannelPhase<SP>>,
```

Comment:
See other comment about moving `ChannelContext` here. Seems we shouldn't need to use an `Option`.

Reply:
I agree that logically it should not be Option, but I found no other way to transport the context from one phase to another. This may be Rust technicality. Option can be taken out and left empty, which is needed as I was not able to make the transfer in one atomic operation. Maybe another solution would be if ChannelContext had a `default()`.

Reply:
Would `swap` work?